### PR TITLE
IDNECHO-2500 - Add test to verify password grant type

### PR DESCRIPTION
**Pull request by diego-desani-sp**
### Changes
- Reorganizing file structure and making some features reusable.
- Adding Org username and key: this should be the same for all dev orgs as far as I understood.
    - We can also create another work item to only pass this org username and key and retrieve username, password and the API clientId and key.
- New feature to validate `Oathkeeper` supported grant types.

```bash
.
├── api
│   ├── ApiTest.java
│   ├── internal
│   │   ├── health-checks.feature
│   │   ├── internal-checks.feature
│   │   └── rate-limit-checks.feature
│   ├── oauth
│   │   └── grant_types.feature
│   ├── v1
│   │   └── basic-v1-apis.feature
│   ├── v2
│   │   └── basic-v2-apis.feature
│   └── v3
│       └── client-credentials-flow.feature
├── authentication
│   ├── basic-auth.js
│   └── oauth
│       ├── cc
│       │   ├── password-hash.js
│       │   └── password_grant_type.feature
│       └── oathkeeper
│           ├── client_credentials_grant_type.feature
│           └── password_grant_type.feature
└── karate-config.js
```

image

### CODE REVIEW

File: src/test/java/authentication/basic-auth.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
-function basicAuth() {
-	var temp = karate.get('apiUser') + ':' + karate.get('apiKey');
+function basicAuth(creds) {
+	karate.log("creds: " + creds)
```
Comment (jeff-upton-sp) [inline src/test/java/authentication/basic-auth.js]: debug code?

File: src/test/java/karate-config.js [hunk 2]
```diff
@@ -4,6 +4,8 @@ function config() {
 	var DEFAULT_API_KEY = '4582c8dkT2x839sypaPtke3';
 	var DEFAULT_USERNAME = 'support';
 	var DEFAULT_PASSWORD = '2thecloud';
+	var DEFAULT_ORG_USERNAME = '39xU32x5hT24';
+	var DEFAULT_ORG_KEY = '4582c8dkT2x839sypaPtke3';
```
Comment (jeff-upton-sp) [inline src/test/java/karate-config.js]: Why is this different than the DEFAULT_API_KEY above? Do we really need to support running the e2e tests with a client that isn't the org api user/key?

File: src/test/java/karate-config.js [hunk 3]
```diff
@@ -19,6 +21,8 @@ function config() {
 		apiUser: readProperty('org.api-user', DEFAULT_API_USER),
 		apiKey: readProperty('org.api-key', DEFAULT_API_KEY),
 		username: readProperty('org.username', DEFAULT_USERNAME),
-		password: readProperty('org.password', DEFAULT_PASSWORD)
+		password: readProperty('org.password', DEFAULT_PASSWORD),
+		orgApiUser: readProperty('org.api-user', DEFAULT_ORG_USERNAME),
+		orgApiKey: readProperty('org.api-user', DEFAULT_ORG_KEY)
```
Comment (jeff-upton-sp) [inline src/test/java/karate-config.js]: Same as above - shouldn't this just be the apiUser/apiKey above? We can require that it is the org api user/key for these tests.

File: src/test/java/karate-config.js (hunk 2 above)
Comment (diego-desani-sp) [reply src/test/java/karate-config.js]: That was temp code change. I'm revering that.

**Pull request merged**